src/libraries/System.Diagnostics.EventLog/src/System/Diagnostics/Reader/NativeWrapper.cs

```diff
@@ -985,11 +987,12 @@ public static IEnumerable<string> EvtFormatMessageRenderKeywords(EventLogHandle
         {
             IntPtr buffer = IntPtr.Zero;
             int bufferNeeded;
+            char[] emptyBuffer = { '\0' }; // issue: https://github.com/dotnet/runtime/issues/100198
 
             try
             {
                 List<string> keywordsList = new List<string>();
-                bool status = UnsafeNativeMethods.EvtFormatMessageBuffer(pmHandle, eventHandle, 0, 0, IntPtr.Zero, flag, 0, IntPtr.Zero, out bufferNeeded);
```

Comment:
Why was this changed?

Reply:
The overload for `EvtFormatMessageBuffer` takes an `IntPtr` but now I'm calling the character version since I chose not to allocate the one character on the heap with `AllocHGlobal` as is done below.

I suppose this code is using `IntPtr` because it wants to just allocate the larger block of memory and parse things out with `PtrToStringUni` rather than slicing strings out of a character array.  Probably it's a side effect of previously this code used `StringBuilder` and that wouldn't handle pulling multiple strings out of a single buffer.  I think we could rewrite the below code to use a `char[]` and `slice` instead and save some use of `Marshal` but I'm not inclined to do that in this bugfix.